runtime/near-evm-runner/src/lib.rs

```diff
@@ -354,12 +402,18 @@ pub fn run_evm(
                 used_gas: context.gas_counter.used_gas(),
                 logs: context.logs,
             };
-            (Some(outcome), None)
-        }
-        Err(VMLogicError::EvmError(err)) => {
-            (None, Some(VMError::FunctionCallError(FunctionCallError::EvmError(err))))
+            (Some(outcome), None, context.evm_gas_counter.used_gas)
```

Comment:
Hm, I'm curious how is this done in other places? Seems like this is wrong level where evm gas should be surfaced.
And the normal gas is in the VMOutcome

Reply:
run_evm is only called at two places, one is actual call evm in vm runner:
![image](https://user-images.githubusercontent.com/13259400/93242667-03149380-f73c-11ea-8e69-14f3cc291cc7.png)

where i'm going to convert evm gas and add into near gas (it could be also add somewhere else, in that case, this return arg is ignored)

Another is in runtime_param_estimator:
![image](https://user-images.githubusercontent.com/13259400/93242965-66062a80-f73c-11ea-85a5-f311e7739364.png)

where we're only interested in evm_gas used, and are measuring number of cpu instructions and calculate a relationship between evm_gas used vs cpu instructions (cost).

> how is this done in other places? 
In near gas estimator, there is a feature `costs_counting` turn on measure mode and it's turned off in production, turned on in param estimator. but it's not necessary here, because evm crate will always calculate you evm gas used, and it's not expensive to return one more argument. In the case you don't need, you can ignore it.